### PR TITLE
Update the pre-commit and github workflow to move pylint to pre-commit.

### PR DESCRIPTION
**Pull request by lesamouraipourpre**
Also, disable duplicate-code check in examples.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -17,3 +17,18 @@ repos:
...
+        name: pylint (library code)
+        types: [python]
+        exclude: "^(docs/|examples/|setup.py$)"
+-   repo: local
+    hooks:
+    -   id: pylint_examples
+        name: pylint (examples code)
+        description: Run pylint rules on "examples/*.py" files
+        entry: /usr/bin/env bash -c
+        args: ['([[ ! -d "examples" ]] || for example in $(find . -path "./examples/*.py"); do pylint --disable=missing-docstring,invalid-name,duplicate-code $example; done)']
```
Comment (dherrada) [inline .pre-commit-config.yaml:33]: Hi @lesamouraipourpre Thanks for doing this. Do you think you could try to copy what we've got [here](https://github.com/adafruit/cookiecutter-adafruit-circuitpython/blob/master/%7B%25%20if%20cookiecutter.library_prefix%20%25%7D%7B%7B%20cookiecutter.library_prefix%20%7C%20capitalize%20%7D%7D_%7B%25%20endif%20%25%7DCircuitPython_%7B%7B%20cookiecutter.library_name%7D%7D/.pre-commit-config.yaml). I think we found another way of getting around the issues with the duplicate-code check without disabling it.

**Review by dherrada — CHANGES_REQUESTED** [review]

**Comment by dherrada** [comment]
@lesamouraipourpre Can you check that the changes you're making are the same as the ones in the place I linked? I think they are, but I just wanted you to check as well before I merge.

**Comment by lesamouraipourpre** [comment]
> 
> @lesamouraipourpre Can you check that the changes you're making are the same as the ones in the place I linked? I think they are, but I just wanted you to check as well before I merge.

Yes, it was essentially a copy-and-paste to add the pylint sections.

Confirmed with:
> 
> diff cookiecutter-adafruit-circuitpython/\{%\ if\ cookiecutter.library_prefix\ %\}\{\{\ cookiecutter.library_prefix\ \|\ capitalize\ \}\}_\{%\ endif\ %\}CircuitPython_\{\{\ cookiecutter.library_name\}\}/.pre-commit-config.yaml Adafruit_CircuitPython_GPS/.pre-commit-config.yaml

**Comment by jposada202020** [comment]
@dherrada :) according to what I read I guess we could merge this change. let me know. Thanks!

**Comment by dherrada** [comment]
Sweet. Looks good. @jposada202020 Thanks for the reminder on this one

**Comment by jposada202020** [comment]
No problem :) @dherrada

**Pull request merged**